enhancements/network/routed-ingress-primary-udn-static-ips.md

```diff
@@ -115,17 +115,63 @@ statically in the guest.
 
 ## Proposal
 
-This section should explain what the proposal actually is. Enumerate
-*all* of the proposed changes at a *high level*, including all of the
-components that need to be modified and how they will be
-different. Include the reason for each choice in the design and
-implementation that is proposed here.
-
-To keep this section succinct, document the details like API field
-changes, new images, and other implementation details in the
-**Implementation Details** section and record the reasons for not
-choosing alternatives in the **Alternatives** section at the end of
-the document.
+This proposal to import VMs from other virtualization platforms while
+preserving their networking configuration will impact three different OpenShift
+components:
+- MTV
+- KubeVirt's ipam-extensions
+- OVN-Kubernetes
+
+We will elaborate the overall flow first, before digging into details in each
+of the components.
+
+MTV currently introspects the VM to learn the MAC addresses of the interfaces,
+and templates the VM which will be created in OpenShift Virtualization with the
+required MAC address. We would need MTV to somehow also figure out what IP
+addresses are on the aforementioned interfaces.
+
+A new CRD - named `IPPool`, or `DHCPLeaseConfig` (or the like) - will be
+created, and is associated to a UDN (both UDN, and C-UDN). This CRD holds the
+association of MAC address to IPs for a UDN. When importing the VM into
+OpenShift Virt, MTV will provision / update this object with this information.
+
+The `ipam-extensions` mutating webhook will kick in whenever a virt launcher
+pod is created - it will identify when the VM has a primary UDN attachment
+(already happens today), and will also identify when the pod network attachment
+has a MAC address configuration request.
+It will then access the `IPPool` (or `DHCPLeaseConfig` for the UDN) to extract
```

Comment:
btw this CRD should be at the same "namespace" as IPAMClaim it should not be a OVN thing

Reply:
hm, not 100% sure yet. 

How will it work for C-UDNs ? We need the MAC:IPs association per **network**, which is a non-namespaced object.

Furthermore, I envision associating this pool to the network using the NAD, enabling this feature to work outside OVN-K (i.e. making it a part of k8snetworkplumbingwg, as `IPAMClaims` are).

Reply:
Sorry I mean the kind namespace, not the resource namespace like muiltus.io/IPAMClaim or the like.

Reply:
OK, then I agree. I don't see a reason for this CRD to be added on OVN-K. I think we can get it on k8snetworkplumbing - as IPAMClaim is.